src/auth/EncryptedStorage.ts

```diff
@@ -0,0 +1,70 @@
+import ExtendableError from 'es6-error'
+import {useMutation, UseMutationOptions, useQuery, UseQueryOptions} from 'react-query'
+
+import {decryptData, encryptData} from '../legacy/commonUtils'
+import storage from '../legacy/storage'
+import {YoroiWallet} from '../yoroi-wallets'
+
+export class CredentialsNotFound extends ExtendableError {}
+
+export const useReadRootKey = (
+  {id, password}: {id: YoroiWallet['id']; password: string},
+  options?: UseQueryOptions<string, Error>,
+) => {
+  const query = useQuery({
+    enabled: false,
+    retry: false,
+    queryKey: ['rootKey'],
+    queryFn: () => EncryptedStorage.read(StorageKeys.rootKey(id), password),
+    ...options,
+  })
+
+  return {
+    readRootKey: query.refetch,
+    ...query,
+  }
+}
+
+export const useWriteRootKey = (
+  {id}: {id: YoroiWallet['id']},
+  options?: UseMutationOptions<void, Error, {password: string; rootKey: string}>,
+) => {
+  const mutation = useMutation({
+    ...options,
+    mutationFn: ({password, rootKey}) => EncryptedStorage.write(StorageKeys.rootKey(id), rootKey, password),
+  })
+
+  return {
+    writeRootKey: mutation.mutate,
+    ...mutation,
+  }
+}
+
+type StorageKey = `/keystore/${string}-MASTER_PASSWORD`
+export const StorageKeys = {
```

Comment:
`StorageKeys` -> `EncryptedStorageKeys`?

Reply:
if the only key ever used with`EncryptedStorage.read` is `rootKey(wallet.id)`, should it accept `wallet` as the first argument and calculate the key inside the function?

Reply:
or initialize each wallet with an instance of EncryptedStorage that is already prefixed with the wallet id?

Reply:
`wallet.encryptedStorage.read("rootKey", password)`
`wallet.encryptedStorage.write("rootKey", password)`